### PR TITLE
[bootstrap-fileinput]: Add some missing definitions, especially around FileActionSettings

### PR DESCRIPTION
**Pull request by zshall**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://plugins.krajee.com/file-input/plugin-options
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### Changes I've made in this PR:

* Fixed capitalization: `fileActionsettings` => `fileActionSettings`
* Made all file action settings parameters optional.
* Added the following parameters to file action settings:
    * `showUpload`
    * `showDownload`
    * `showRemove`
    * `showZoom`
    * `showDrag`
    * `uploadRetryIcon`
    * `downloadIcon`
    * `downloadClass`
    * `downloadTitle`
    * `zoomIcon`
    * `zoomClass`
    * `zoomTitle`
    * `dragIcon`
    * `dragClass`
    * `dragTitle`
    * `dragSettings`
    * `indicatorPaused`
    * `indicatorPausedTitle`
* Added the following top level parameters for `FileInputOptions`:
    * `required`
    * `rtl`
    * `hideThumbnailContent`
    * `encodeUrl`
    * `focusCaptionOnBrowse`
    * `focusCaptionOnClear`

### CODE REVIEW

File: types/bootstrap-fileinput/index.d.ts [hunk 1]
```diff
@@ -119,6 +120,44 @@ declare namespace BootstrapFileInput {
          * Theming
          */
         theme?: string;
+        /**
+         * Whether file selection is mandatory before upload (for ajax) or submit of the form (for non-ajax). When set to true, and if files are not
+         * selected before upload, this will show the error message as set in msgFilerequired. Defaults to false.
+         */
+        required?: boolean;
```
Comment (peterblazejewicz) [inline types/bootstrap-fileinput/index.d.ts:127]: can I propose, optional change (here and below)
```ts
* @default false
```

File: types/bootstrap-fileinput/index.d.ts [hunk 2]
```diff
@@ -781,7 +820,7 @@ declare namespace BootstrapFileInput {
         /**
          * configuration for setting up file actions for newly selected file thumbnails in the preview window.
          */
-        fileActionsettings?: FileActionSettings;
```
Comment (peterblazejewicz) [inline types/bootstrap-fileinput/index.d.ts:784]: 👁️

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@zshall thx for the change!

File: types/bootstrap-fileinput/index.d.ts (hunk 1 above)
Comment (zshall) [reply types/bootstrap-fileinput/index.d.ts:127]: Sounds good! I'll add those.

Comment (zshall) [reply types/bootstrap-fileinput/index.d.ts:128]: @peterblazejewicz I've added `@default` values for the booleans you mentioned and replaced the comments `Defaults to false` and `Defaults to true` with `@default false` and `@default true` where applicable. Also added `@default`s for the booleans in `FileActionSettings` based on the [docs](https://plugins.krajee.com/file-input/plugin-options#fileActionSettings).

**Pull request merged**